### PR TITLE
fix(respect): secrets masking for encoded occurrences in har-output

### PR DESCRIPTION
**Pull request by DmitryAnansky**
## What/Why/How?

Fixed secrets masking to cover encoded occurrences of a secret in har-output.

## Check yourself

- [x] This PR follows the [contributing guide](https://github.com/Redocly/redocly-cli/blob/main/CONTRIBUTING.md#pull-request-guidelines) 
- [x] All new/updated code is covered by tests
- [ ] Core code changed? - Tested with other Redocly products (internal contributions only)
- [ ] New package installed? - Tested in different environments (browser/node)
- [x] Documentation update has been considered 

## Security

- [x] The security impact of the change has been considered 
- [x] Code follows company security practices and guidelines

---

> [!NOTE]
> **Low Risk**
> Localized improvement to secret redaction in logging/HAR output with broad test coverage; no auth or data-path changes beyond safer masking.
> 
> **Overview**
> **Secrets masking** in `@redocly/respect-core` now replaces encoded forms of known secrets, not only literal substrings—so values like URL-encoded form bodies, JSON-escaped strings, and nested HAR request/response text are redacted in `har-output`.
> 
> `maskSecrets` builds a **pattern list** per secret (raw value, JSON string escape body, `encodeURIComponent`, and `URLSearchParams` form style), skips whitespace-only entries, avoids `encodeURIComponent` on ill-formed strings (lone surrogates), and applies **longest-match-first** so a shorter secret cannot leave part of a longer one visible. Tests cover HAR captures, multiple occurrences in one string, and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit acf399376b3b8ccc27e6bd93496cd421a321c131. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by tatomyr — APPROVED** [review]

**Comment by adamaltman** [comment]
What about the `proxy` command?

**Comment by DmitryAnansky** [comment]
> What about the `proxy` command?

The `proxy` command captures live HTTP traffic through a reverse proxy into a HAR file, and optionally validates it against an OpenAPI description in real time, but does no secret masking.

`respect` knows the secret values (it injects them itself), while `proxy` passively records someone else's traffic and has no secrets list.

cc: @sobanieca-redocly , @Redocly/cyberpunks

File: .changeset/mask-encoded-secrets.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@redocly/respect-core': patch
+---
+
+Fixed secrets masking to cover encoded occurrences of a secret in har-output.
```
Comment (JLekawa) [inline .changeset/mask-encoded-secrets.md]:
```suggestion
Fixed an issue where secrets masking did not cover encoded secrets in `har-output`.
```

**Review by JLekawa — APPROVED** [review]
